docs/wiki/framework/wardrobe-framework.md

```diff
@@ -0,0 +1,300 @@
+version:
+  major: 3
+  minor: 20
+  patch: 0
+---
+
+
+## 1. Overview
+
+The Wardrobe Addon gives the player the opportunity to change/modify their current wearables into different variants.
+
+For example, if a uniform has a normal variant and a "Rolled-Up Sleeves" variant, the player will be able to use ether an ACE Self Interaction or the CBA Context Menu in the Inventory.
+
+Since there is no reliable, common pattern in terms of class inheritence, not even within the same DLC, each possbile variant has to be defined individually within the classes config properties.
+
+### 1.1 Components
+
+Components are an optional assistance to define a group of variants where a variant is made of one or more components.
+For example, a bandanna with sunglasses would have the the bandanna and the sunglasses as components, while the bandanna and sunglasses themself only have themselves as a component.
+
+If the player switches from a variant with more components to a variant with less components, the surplus components will be deposited in their inventory. Vise Versa, if the player wants to change from a variant with less components to a variant with more components, they are required to have the missing components in their inventory, which will be removed on conversion.
+
+More examples can be found below.
+
+### 1.2 Config Guidelines
+
+- An Container (Uniform, Vest, Backpack) should never change its maxLoad, unless there is a really good reason for it. This will result in the loss of items.
+- Variants should never "magically" add/remove parts of themselves, unless its handled through a component.
+- There should not be a difference in weight between the different variants unless compoents get added/removed.
+  - For Example: `Mass of Helmet with Goggles = Mass of Helmet + Mass of Goggles.`
+
+## 2. Config
+
+Only directly defined Subclasses will be taken into account. A fully inherited subclass will be ignored and will not provide any interaction. The ace_wardrobe properties are to be found as a subclass of the item itself. Multiple base configs are designed at root of `configFile`.
+
+## 2.1 Properties
+
+
+| Class Property |  Data Type | Description |
+| -------------- |  ----------- | ----------- |
+| modifiableTo[] | Array of Classnames | possible Variants this item can be turned into. |
+| components[] | Array of Classnames  | Components the current variant contains within itself | 
+| sound[]  | Array of CfgSound Entries | to be chosen by random when the action is performed |
+| sound_timing | Number 0..1 | defines the point time relative to the duration when the sound is played |
+| gesture | String of Classname | Gesture to be played when the action is performed |
+| alternativePicture | String of path to icon | to be used instead of target variant picture |
+| alternativeDisplayName | String | Will be used instead of the target variants displayname |
+| duration | Number in seconds | Duration of action. Items are being replaced at the end. |
```

Comment:
```suggestion
| `modifiableTo[]` | Array of Classnames | possible Variants this item can be turned into. |
| `components[]` | Array of Classnames  | Components the current variant contains within itself | 
| `sound[]` | Array of CfgSound Entries | to be chosen by random when the action is performed |
| `sound_timing` | Number 0..1 | defines the point time relative to the duration when the sound is played |
| `gesture` | String of Classname | Gesture to be played when the action is performed |
| `alternativePicture` | String of path to icon | to be used instead of target variant picture |
| `alternativeDisplayName` | String | Will be used instead of the target variants displayname |
| `duration` | Number in seconds | Duration of action. Items are being replaced at the end. |
```